src/features/Common/data/api.js

```diff
@@ -0,0 +1,19 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { getConfig } from '@edx/frontend-platform';
+
+function getClassesByInstructor(instructorUsername, limit = false, page = '', urlParamsFilters = '') {
```

Comment:
The function name Itself is clear about we're working with instructors so I think `instructorUserName` is redundant, why just leave `userNmae` , in this context is very clear itself, don't you?

```suggestion
function getClassesByInstructor(userName, limit = false, page = '', urlParamsFilters = '') {
```
<br/>
<br/>

Uhmmm, now I got a few mixed thoughts about the optional parameters its becomes a truthfully headache at the moment to handle or add several arguments how has it happened in IP.
![image](https://github.com/user-attachments/assets/17a6bcde-e1da-4b96-b6e7-b39b14ec661d)

Let's picture this scenario, I pretty sure this endpoint could be called multi purpose endpoint so, for whatever the reason now I need to add `classId` to get results by classId so if add this extra param to the implementation It will look like:
```js
function getClassesByInstructor(instructorUsername, limit = false, page = '', urlParamsFilters = '', classId="") {
```
And at the moment to call it:

```js
 getClassesByInstructor('Sam', null, null, null, "example-id+123") 
```
Don't u think is kind weird to pass all bunch of `null` in the function? What if  we must add another parameter for any reason? The optional parameters will be growing and growing and makes more difficult track which value referrers to who, so what if we implement a default config pattern, in practice it look like:

```js
function getClassesByInstructor(userName, options = {}) {
 // This could be a general const, this settings will be used in more places.
  const defaultOptions = {
    limit: false,
    page: '',
    urlParamsFilters: {},
  };

  const params = {
    ...defaultOptions,
    ...options,
    instructor: userName,
    ...options.urlParamsFilters,
  };

  return getAuthenticatedHttpClient().get(
    `${getConfig().COURSE_OPERATIONS_API_V2_BASE_URL}/classes/`,
    { params },
  );
}
```

With this approach now is more flexible to send the only stuff that we need, what do u think?


Finally I strongly recommend to add jsdocs for this kind of functions to make more clear what means all paramethers, for instance `limit` is not very clear, but with documentation any pearson could realice to this param determines if we're using pagination or not.

Here I leave an example but feel free to improve it.

```js
 * Fetches classes by instructor username with optional filters and pagination.
 *
 * @param {string} username - The username of the instructor.
 * @param {Object} [options={}] - Optional parameters.
 * @param {boolean} [options.limit=false] - Determines if the pagination is required.
 * @param {string} [options.page=''] - The page number for page pagination.
 * @param {Object} [options.urlParamsFilters={}] - Additional URL parameters for filtering results.
 * @param {string} [options.urlParamsFilters.courseType] - Filter by course type.
 * @param {string} [options.urlParamsFilters.language] - Filter by language.
 * 
 * @returns {Promise} - A promise that resolves to the response from the API.
 */
```